File: CS/DXSample/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Media;

namespace DXSample.ViewModel {
    [POCOViewModel]
    public class MainViewModel {
        public virtual int Progress { get; protected set; }
        public virtual Brush Color { get; protected set; }
        public virtual bool IsProgress { get; protected set; }
        protected IDispatcherService DispatcherService { get { return this.GetService<IDispatcherService>(); } }
        public void Calculate() {
            IsProgress = true;
            Task.Factory.StartNew(CalcCore).ContinueWith(x => {
                DispatcherService.BeginInvoke(() => IsProgress = false);
            });
        }
        protected void OnIsProgressChanged() {
            this.RaiseCanExecuteChanged(x => x.Calculate());
        }

        public bool CanCalculate() {
            return !IsProgress;
        }
        void CalcCore() {
            for (int i = 1; i <= 100; i++) {
                DispatcherService.BeginInvoke(() => {
                    Progress = i;
                    Color = new SolidColorBrush(GetColor(i));
                });
                Thread.Sleep(TimeSpan.FromSeconds(0.1));
            }
        }
        Color GetColor(int coef) {
            return new Color() {
                R = (byte)(coef * 15),
                G = (byte)(coef * 7),
                B = (byte)(coef * 8),
                A = 255,
            };
        }
    }
}

File: VB/DXSample/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.DataAnnotations
Imports System
Imports System.Threading
Imports System.Threading.Tasks
Imports System.Windows.Media

Namespace DXSample.ViewModel

    <POCOViewModel>
    Public Class MainViewModel

        Public Overridable Property Progress As Integer

        Public Overridable Property Color As Brush

        Public Overridable Property IsProgress As Boolean

        Protected ReadOnly Property DispatcherService As IDispatcherService
            Get
                Return GetService(Of IDispatcherService)()
            End Get
        End Property

        Public Sub Calculate()
            IsProgress = True
            Call Task.Factory.StartNew(AddressOf CalcCore).ContinueWith(Sub(x)
                DispatcherService.BeginInvoke(Sub() IsProgress = False)
            End Sub)
        End Sub

        Protected Sub OnIsProgressChanged()
            RaiseCanExecuteChanged(Sub(x) x.Calculate())
        End Sub

        Public Function CanCalculate() As Boolean
            Return Not IsProgress
        End Function

        Private Sub CalcCore()
            For i As Integer = 1 To 100
                DispatcherService.BeginInvoke(Sub()
                    Progress = i
                    Color = New SolidColorBrush(GetColor(i))
                End Sub)
                Thread.Sleep(TimeSpan.FromSeconds(0.1))
            Next
        End Sub

        Private Function GetColor(ByVal coef As Integer) As Color
            Return New Color() With {.R = CByte(coef * 15), .G = CByte(coef * 7), .B = CByte(coef * 8), .A = 255}
        End Function
    End Class
End Namespace
